### PR TITLE
share functionality

### PR DESCRIPTION
**Pull request by Karlmusingo**
#### What does this PR do?
Save who shared, what they saved, through which plateforme(facebook, twitter and mail)
#### Description of Task to be completed?
`POST /posts/:slug/share/:plateforme`
#### How should this be manually tested?
- Clone the repo
- Run npm run test
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

**Comment by Karlmusingo** [comment]
[Posts should be shareable](https://trello.com/c/5PfKYJnt/18-posts-should-be-shareable)

File: src/middlewares/checkPostAndJob.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { Post, Job } from '../models';
+import * as statusCodes from '../constants/statusCodes';
+import { notExist } from '../constants/responseMessages';
+
+const checkPostAndJob = async (req, res, next) => {
```
Comment (oesukam) [inline src/middlewares/checkPostAndJob.js:5]: @Karlmusingo  Why this middleware when we already have `checkPost` and `checkJob`

File: src/models/Share.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+const ShareSchema = new Schema({
+  plateforme: {
+    type: String,
+  },
+  user: {
+    type: Schema.Types.ObjectId,
+    ref: 'User',
+  },
+  post: {
```
Comment (oesukam) [inline src/models/Share.js:11]: @Karlmusingo  Job and Post reference should be combined into one and have another property to differentiate them

File: src/routes/api/v1/posts.js [hunk 3]
```diff
@@ -26,4 +31,14 @@ router
...
 
+router
+  .route('/:postSlug/share/:plateforme')
+  .post(
+    checkAuth,
+    celebrate({ params: postValidator.sharePost }),
+    checkPostAndJob,
+    asyncHandler(PostController.sharePost),
+  );
+// .get(asyncHandler(PostController.getPosts));
```
Comment (oesukam) [inline src/routes/api/v1/posts.js]: @Karlmusingo Whis this comment?

**Review by oesukam — CHANGES_REQUESTED** [review]

File: src/middlewares/checkPostAndJob.js (hunk 1 above)
Comment (Karlmusingo) [reply src/middlewares/checkPostAndJob.js:5]: I didn't want to modify the other middlewares and I needed to check in the two models before returning the response if none of Jobs or Posts is found or doing next

File: src/models/Share.js (hunk 2 above)
Comment (Karlmusingo) [reply src/models/Share.js:11]: I did that because I wanted to put the reference for both Posts and Jobs

**Pull request merged**